core/src/main/scala/kafka/migration/MigrationPropagator.scala

```diff
@@ -118,9 +124,8 @@ class MigrationPropagator(
       }
     }
 
-    // If there are new brokers (including KRaft brokers) or if there are changes in topic
-    // metadata, let's send UMR about the changes to the old Zk brokers.
-    if (brokersChanged || !delta.topicsDelta().deletedTopicIds().isEmpty || !delta.topicsDelta().changedTopics().isEmpty) {
+    // If there are changes in topic metadata, let's send UMR about the changes to the old Zk brokers.
+    if (!delta.topicsDelta().deletedTopicIds().isEmpty || !delta.topicsDelta().changedTopics().isEmpty) {
```

Comment:
Sorry, this might be a silly question, but is it OK for `requestBatch.addUpdateMetadataRequestForBrokers` to be called twice in some cases?

Reply:
Yea, it should be fine since the request batch is using a set